openscm/adapters/dice.py

```diff
@@ -0,0 +1,388 @@
+            ParameterType.AVERAGE_TIMESERIES,
+        )
+
+        # Total CO2 emissions (GtCO2 per year)
+        self._views.E = self._parameters.get_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtCO2/a",
+            time_points_for_averages,
+            ParameterType.AVERAGE_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # TODO remove setting missing emissions once there is a method in tests for that
+        if self._views.E.is_empty:
+            self._parameters.get_writable_timeseries_view(
+                ("Emissions", "CO2"),
+                ("World",),
+                "GtCO2/a",
+                time_points_for_averages,
+                ParameterType.AVERAGE_TIMESERIES,
+                InterpolationType.LINEAR,
+                ExtrapolationType.LINEAR,
+            ).set(np.zeros(self._timestep_count))
+
+        # Concentration in atmosphere
+        self._views.M_atm = self._output.get_writable_timeseries_view(
+            ("Concentration", "Atmosphere"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Carbon concentration increase in lower oceans (rel. to 1750)
+        self._views.M_l = self._output.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "lower"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Carbon concentration increase in shallow oceans (rel. to 1750)
+        self._views.M_u = self._output.get_writable_timeseries_view(
+            ("Concentration", "Ocean", "shallow"),
+            ("World",),
+            "GtC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Increase temperature of atmosphere (rel. to 1900)
+        self._views.T_atm = self._output.get_writable_timeseries_view(
+            ("Temperature Increase", "Atmosphere"),
+            ("World",),
+            "degC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Increase in temperature of lower oceans (rel. to 1900)
+        self._views.T_ocean = self._output.get_writable_timeseries_view(
+            ("Temperature Increase", "Ocean", "lower"),
+            ("World",),
+            "degC",
+            time_points,
+            ParameterType.POINT_TIMESERIES,
+            InterpolationType.LINEAR,
+            ExtrapolationType.LINEAR,
+        )
+
+        # Increase in radiative forcing (rel. to 1900)
```

Comment:
is radiative forcing actually relative in DICE?

Reply:
yes